### PR TITLE
[zh]update /docs/tasks/extensibility/wasm-module-distribution/index.md

### PR DESCRIPTION
**Pull request by yanrongshi**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+---
+
+Istio 提供了[使用 WebAssembly（Wasm）扩展代理功能](/zh/blog/2020/wasm-announce/)的能力。
+Wasm 可扩展性的关键优势之一是扩展可以在运行时动态加载。
+这些扩展必须首先分发到 Envoy 代理。
+Istio 通过允许代理动态下载 Wasm 模块来实现这一点。
+
+## 安装测试应用程序{#setup-the-test-application}
+
+在你开始这项任务之前，请部署[Bookinfo](/docs/examples/bookinfo/#deploying-the-application)示例应用程序。
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:22]:
```suggestion
在你开始这项任务之前，请部署[Bookinfo](/zh/docs/examples/bookinfo/#deploying-the-application)示例应用程序。
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+{{< /text >}}
+
+1. 带凭据测试 `/productpage`
+
+{{< text bash >}}
+$ curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
+200
+{{< /text >}}
+
+关于 `WasmPlugin` API 的更多使用示例，请查看 [API 参考](/docs/reference/config/proxy_extensions/wasm-plugin/)。
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:78]:
```suggestion
关于 `WasmPlugin` API 的更多使用示例，请查看 [API 参考](/zh/docs/reference/config/proxy_extensions/wasm-plugin/)。
```

File: content/zh/docs/tasks/extensibility/_index.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+title: Extensibility
```
Comment (kebe7jun) [inline content/zh/docs/tasks/extensibility/_index.md:2]: 翻译一下？可扩展性？

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
+---
+title: 分发 WebAssembly 模块
```
Comment (kebe7jun) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:2]:
```suggestion
title: WebAssembly 模块分发
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+description: 描述如何使远程 WebAssembly 模块在网格中可用。
```
Comment (kebe7jun) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:3]:
```suggestion
description: 描述如何在网格中使用远程 WebAssembly 模块。
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+这些扩展必须首先分发到 Envoy 代理。
+Istio 通过允许代理动态下载 Wasm 模块来实现这一点。
+
+## 安装测试应用程序{#setup-the-test-application}
+
+在你开始这项任务之前，请部署[Bookinfo](/zh/docs/examples/bookinfo/#deploying-the-application)示例应用程序。
+
+## 配置 Wasm 模块{#configure-wasm-modules}
+
+在这个例子中，你将在你的网格中添加一个 HTTP Basic 身份验证扩展。您将配置 Istio 从远程映像注册表中提取[基本身份验证模块](https://github.com/istio-ecosystem/wasm-extensions/tree/master/extensions/basic_auth)并加载它。它将被配置为在对 `/productpage` 调用中运行。
```
Comment (kebe7jun) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:26]: 您。

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+          - "ok:test"
+          - "YWRtaW4zOmFkbWluMw=="
+EOF
+{{< /text >}}
+
+HTTP 过滤器将作为身份验证过滤器注入到入口网关代理中。
+Istio 代理将解释 `WasmPlugin` 配置，从 OCI 图像注册表中下载远程 Wasm 模块到本地文件，并通过引用该文件将 HTTP 过滤器注入 Envoy 中。
+
+{{< idea >}}
+如果在 `istio-system` 之外的特定命名空间中创建了 `WasmPlugin`，则该命名空间中的 pods 将被配置。如果在 `istio-system` 命名空间中创建资源，所有命名空间都会受到影响。
```
Comment (kebe7jun) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:59]:
```suggestion
如果在 `istio-system` 之外的特定命名空间中创建了 `WasmPlugin`，则该命名空间中的 Pod 将被配置。如果在 `istio-system` 命名空间中创建资源，所有命名空间都会受到影响。
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+关于 `WasmPlugin` API 的更多使用示例，请查看 [API 参考](/zh/docs/reference/config/proxy_extensions/wasm-plugin/)。
+
+## 清理 Wasm 模块{#clean-up-wasm-modules}
+
+{{< text bash >}}
+$ kubectl delete wasmplugins.extensions.istio.io -n istio-system basic-auth
+{{< /text >}}
+
+## 监控 Wasm 模块分布{#monitor-wasm-module-distribution}
```
Comment (kebe7jun) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:86]:
```suggestion
## 监控 Wasm 模块分发{#monitor-wasm-module-distribution}
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+有几个统计数据可以跟踪远程 Wasm 模块的分布状态。
```
Comment (kebe7jun) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:88]: 分发。

File: content/zh/docs/tasks/extensibility/_index.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+description: Demonstrates how to extend mesh behavior.
```
Comment (windsonsea) [inline content/zh/docs/tasks/extensibility/_index.md:3]:
```suggestion
description: 演示如何扩展网格行为。
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md (hunk 6 above)
Comment (windsonsea) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:26]:
```suggestion
在这个例子中，您将在您的网格中添加一个 HTTP Basic 身份验证扩展。您将配置 Istio 从远程镜像仓库中提取并加载[基本身份验证模块](https://github.com/istio-ecosystem/wasm-extensions/tree/master/extensions/basic_auth)。该模块将被配置为在调用到 `/productpage` 时运行。
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+          - "GET"
+          - "POST"
+        credentials:
+          - "ok:test"
+          - "YWRtaW4zOmFkbWluMw=="
+EOF
+{{< /text >}}
+
+HTTP 过滤器将作为身份验证过滤器注入到入口网关代理中。
+Istio 代理将解释 `WasmPlugin` 配置，从 OCI 图像注册表中下载远程 Wasm 模块到本地文件，并通过引用该文件将 HTTP 过滤器注入 Envoy 中。
```
Comment (windsonsea) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:56]:
```suggestion
Istio 代理将解释 `WasmPlugin` 配置，从 OCI 镜像仓库中下载远程 Wasm 模块到本地文件，并通过引用该文件将 HTTP 过滤器注入 Envoy 中。
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+{{< /idea >}}
+
+## 检查配置的 Wasm 模块{#check-the-configured-wasm-module}
+
+1. 不带凭据测试 `/productpage`
+
+{{< text bash >}}
+$ curl -s -o /dev/null -w "%{http_code}" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
+401
+{{< /text >}}
```
Comment (windsonsea) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:69]: Line 66-69 need indentation with 3 spaces.

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+
+1. 带凭据测试 `/productpage`
+
+{{< text bash >}}
+$ curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
+200
+{{< /text >}}
```
Comment (windsonsea) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:76]: Line 73-76 need indentation with 3 spaces.

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+- `istio_agent_wasm_cache_lookup_count`: Wasm 远程获取缓存查找的次数。
+- `istio_agent_wasm_cache_entries`: Wasm 配置转换和结果的数量，包括成功、没有远程加载、编组失败、远程获取失败和未收到远程获取提示。
+- `istio_agent_wasm_config_conversion_duration_bucket`: istio-agent 在 Wasm 模块的配置转换上花费的总时间（以毫秒为单位）。
+- `istio_agent_wasm_remote_fetch_count`: Wasm 远程获取和结果的数量，包括成功、下载失败和校验和不匹配。
+
+如果由于下载失败或其他原因而拒绝了 Wasm 过滤器配置，则 istiod 也会发出带有类型标签 `type.googleapis.com/envoy.config.core.v3.TypedExtensionConfig` 的 `pilot_total_xds_rejects` 。
+
+## 开发 Wasm 扩展{#develop-a-wasm-extension}
+
+要了解关于 Wasm 模块开发的更多信息， 请参阅此存储库中提供的指南 [`istio-ecosystem/wasm-extensions` 存储库](https://github.com/istio-ecosystem/wasm-extensions),
```
Comment (windsonsea) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:101]:
```suggestion
要了解关于 Wasm 模块开发的更多信息，请参阅 [`istio-ecosystem/wasm-extensions` 存储库](https://github.com/istio-ecosystem/wasm-extensions) 中提供的那些指南，
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+由 Istio 社区维护，用于开发 Istio 的 Telemetry Wasm 扩展：
```
Comment (windsonsea) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:102]:
```suggestion
这个存储库由 Istio 社区维护，用于开发 Istio 的 Telemetry Wasm 扩展：
```

File: content/zh/docs/tasks/extensibility/_index.md (hunk 3 above)
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/_index.md:2]:
```suggestion
title: 可扩展性
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+
+{{< idea >}}
+如果在 `istio-system` 之外的特定命名空间中创建了 `WasmPlugin`，则该命名空间中的 Pod 将被配置。如果在 `istio-system` 命名空间中创建资源，所有命名空间都会受到影响。
+{{< /idea >}}
+
+## 检查配置的 Wasm 模块{#check-the-configured-wasm-module}
+
+1. 不带凭据测试 `/productpage`
+
+{{< text bash >}}
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:66]:
```suggestion
    {{< text bash >}}
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+$ curl -s -o /dev/null -w "%{http_code}" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:67]:
```suggestion
    $ curl -s -o /dev/null -w "%{http_code}" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+401
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:68]:
```suggestion
    401
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+{{< /text >}}
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:69]:
```suggestion
    {{< /text >}}
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+
+1. 带凭据测试 `/productpage`
+
+{{< text bash >}}
+$ curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
+200
+{{< /text >}}
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:76]:
```suggestion
    {{< /text >}}
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+{{< text bash >}}
+$ curl -s -o /dev/null -w "%{http_code}" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
+401
+{{< /text >}}
+
+1. 带凭据测试 `/productpage`
+
+{{< text bash >}}
+$ curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
+200
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:75]:
```suggestion
    200
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 22]
```diff
@@ -0,0 +1,113 @@
...
+
+{{< text bash >}}
+$ curl -s -o /dev/null -w "%{http_code}" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
+401
+{{< /text >}}
+
+1. 带凭据测试 `/productpage`
+
+{{< text bash >}}
+$ curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:74]:
```suggestion
    $ curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 23]
```diff
@@ -0,0 +1,113 @@
...
+1. 不带凭据测试 `/productpage`
+
+{{< text bash >}}
+$ curl -s -o /dev/null -w "%{http_code}" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
+401
+{{< /text >}}
+
+1. 带凭据测试 `/productpage`
+
+{{< text bash >}}
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:73]:
```suggestion
     {{< text bash >}}
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 24]
```diff
@@ -0,0 +1,113 @@
...
+
+## 清理 Wasm 模块{#clean-up-wasm-modules}
+
+{{< text bash >}}
+$ kubectl delete wasmplugins.extensions.istio.io -n istio-system basic-auth
+{{< /text >}}
+
+## 监控 Wasm 模块分发{#monitor-wasm-module-distribution}
+
+有几个统计数据可以跟踪远程 Wasm 模块的分布状态。
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:88]:
```suggestion
有几个统计数据可以跟踪远程 Wasm 模块的分发状态。
```

File: content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md [hunk 25]
```diff
@@ -0,0 +1,113 @@
...
+---
+
+Istio 提供了[使用 WebAssembly（Wasm）扩展代理功能](/zh/blog/2020/wasm-announce/)的能力。
+Wasm 可扩展性的关键优势之一是扩展可以在运行时动态加载。
+这些扩展必须首先分发到 Envoy 代理。
+Istio 通过允许代理动态下载 Wasm 模块来实现这一点。
+
+## 安装测试应用程序{#setup-the-test-application}
+
+在你开始这项任务之前，请部署[Bookinfo](/zh/docs/examples/bookinfo/#deploying-the-application)示例应用程序。
```
Comment (yanrongshi) [inline content/zh/docs/tasks/extensibility/wasm-module-distribution/index.md:22]:
```suggestion
在您开始这项任务之前，请部署[Bookinfo](/zh/docs/examples/bookinfo/#deploying-the-application)示例应用程序。
```

**Review by windsonsea — APPROVED** [review]

**Pull request merged**